### PR TITLE
Fixed miscellaneous issues

### PR DESCRIPTION
**Pull request by DarkmatterVale**
-Added makefile
-updated method declaration
-fixed a number of include statements

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+WARNINGS =
+# Build with debugging symbols if set to TRUE
+SYMBOLS=
+COMPILER_FLAGS=-Woverloaded-virtual -funsigned-bitfields -Wwrite-strings 
+LINKER_FLAGS= -lvorbisenc
+
+## include the makefile-engine
+DEVEL_DIRECTORY := $(shell findpaths -r "makefile_engine" B_FIND_PATH_DEVELOP_DIRECTORY)
+
+include $(DEVEL_DIRECTORY)/etc/makefile-engine
```
Comment (waddlesplash) [inline Makefile]: Please use v2.6 of the template makefile.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+DEFINES=
+#	specify special warning levels
+#	if unspecified default warnings will be used
+#	NONE = supress all warnings
+#	ALL = enable all warnings
+WARNINGS =
+# Build with debugging symbols if set to TRUE
+SYMBOLS=
+COMPILER_FLAGS=-Woverloaded-virtual -funsigned-bitfields -Wwrite-strings 
+LINKER_FLAGS= -lvorbisenc
```
Comment (waddlesplash) [inline Makefile]: Please add this to LIBS and not linker flags.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,29 @@
+NAME= SimplyVorbis
+TYPE= APP
+SRCS= AboutWindow.cpp App.cpp AutoTextControl.cpp CDAudioDevice.cpp CDDBSupport.cpp MainWindow.cpp Preferences.cpp PrefsWindow.cpp RipSupport.cpp RipView.cpp TypedList.cpp
+RSRCS= SimplyVorbis.rsrc
+LIBS= /boot/system/lib/libbe.so /boot/system/lib/libgame.so /boot/system/lib/libmedia.so /boot/system/lib/libnet.so /boot/system/lib/libnetapi.so /boot/system/lib/libroot.so /boot/system/lib/libtextencoding.so /boot/system/lib/libtracker.so /boot/system/lib/libtranslation.so ogg vorbis
```
Comment (waddlesplash) [inline Makefile]: Don't use hardcoded paths.

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+LIBPATHS=
+SYSTEM_INCLUDE_PATHS= /boot/develop/headers/be /boot/develop/headers/cpp /boot/develop/headers/posix /boot/develop/lib /boot/beos/system/lib /vorbis /ogg /
+LOCAL_INCLUDE_PATHS= boot/home/config/lib
```
Comment (waddlesplash) [inline Makefile]: All the above include paths are default, you can delete them.

File: Makefile (hunk 3 above)
Comment (DarkmatterVale) [inline Makefile]: Do I need hardcoded paths, or can I just remove the paths all together without having any problems?

File: Makefile (hunk 4 above)
Comment (DarkmatterVale) [inline Makefile]: See the above comment

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+#		you can simply specify XXX for the library. (e.g. the entry for
+#		"libtracker.so" would be "tracker")
+#
+#	-	for GCC-independent linking of standard C++ libraries, you can use
+#		$(STDCPPLIBS) instead of the raw "stdc++[.r4] [supc++]" library names.
+#
+#	- 	if your library does not follow the standard library naming scheme,
+#		you need to specify the path to the library and it's name.
+#		(e.g. for mylib.a, specify "mylib.a" or "path/mylib.a")
+LIBS = be libgame.so libmedia.so libnet.so libnetapi.so libroot.so libtextencoding.so libtracker.so libtranslation.so ogg vorbis vorbisenc
```
Comment (waddlesplash) [inline Makefile]: Delete all the "lib" and ".so" parts, they aren't needed. Also remove libroot entirely, it's linked in by default. Also, does this application really use libgame, libnet, and libnetapi? I find that hard to believe...

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+#%{
+# @src->@ 
+
+#	Specify the source files to use. Full paths or paths relative to the 
+#	Makefile can be included. All files, regardless of directory, will have
+#	their object files created in the common object directory. Note that this
+#	means this Makefile will not work correctly if two source files with the
+#	same name (source.c or source.cpp) are included from different directories.
+#	Also note that spaces in folder names do not work well with this Makefile.
+SRCS = AboutWindow.cpp App.cpp AutoTextControl.cpp CDAudioDevice.cpp CDDBSupport.cpp MainWindow.cpp Preferences.cpp PrefsWindow.cpp RipSupport.cpp RipView.cpp TypedList.cpp
```
Comment (waddlesplash) [inline Makefile]: Please wrap these lines (the same way lines are wrapped in HaikuPorts).

Comment (DarkmatterVale) [inline Makefile]: How is that done in HaikuPorts?

File: Makefile (hunk 5 above)
Comment (DarkmatterVale) [inline Makefile]: I do not know whether it uses those libraries.....That is what was put in the makefile by Paladin.

Comment (waddlesplash) [inline Makefile]: Then remove one at a time and find out.

File: Makefile (hunk 6 above)
Comment (waddlesplash) [inline Makefile]: Just wrap each line after the .cpp by adding a `\` just before the newline character.

File: Makefile (hunk 5 above)
Comment (diversys) [inline Makefile]: libnet can be removed, libgame and libnetapi are needed.

**Comment by DarkmatterVale** [comment]
Diversys,

When I built it, I did not get any errors when I removed those three libraries. Does it need it not during the BUILD() but is used in the actual application?

**Comment by diversys** [comment]
Doubt it. I wasn't using haikuporter to build it though, just cloned your repo and ran make.

**Comment by DarkmatterVale** [comment]
And that didn't work?

**Comment by diversys** [comment]
No, libnetapi is required for cddb support and libgame is required for BFileGameSound api.
See build log http://pastie.org/pastes/9805009/text

**Comment by DarkmatterVale** [comment]
Interesting....For some reason mine built fine. Is there anything else that should be changed?

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+#		you can simply specify XXX for the library. (e.g. the entry for
+#		"libtracker.so" would be "tracker")
+#
+#	-	for GCC-independent linking of standard C++ libraries, you can use
+#		$(STDCPPLIBS) instead of the raw "stdc++[.r4] [supc++]" library names.
+#
+#	- 	if your library does not follow the standard library naming scheme,
+#		you need to specify the path to the library and it's name.
+#		(e.g. for mylib.a, specify "mylib.a" or "path/mylib.a")
+LIBS = be media root textencoding tracker translation ogg vorbis vorbisenc
```
Comment (waddlesplash) [inline Makefile:59]: `root` is included by default, you can remove it.

Comment (DarkmatterVale) [inline Makefile:59]: Ok. Anything else?

Comment (waddlesplash) [inline Makefile:59]: I can't think of anything, no.

**Pull request merged**